code/datums/components/harvester.dm

```diff
@@ -231,20 +222,18 @@
 			target.apply_status_effect(/datum/status_effect/incapacitating/harvester_slowdown, 1 SECONDS)
 
 		if(/datum/reagent/medicine/kelotane)
+			target.apply_damage(weapon.force*0.6, BRUTE, user.zone_selected)
 			target.adjust_sunder(7.5) //Same amount as a shotgun slug
 			target.flamer_fire_act(10)
-			target.apply_damage(max(0, 20 - 20*target.hard_armor.getRating("fire")), BURN, user.zone_selected, FIRE)
-			var/list/cone_turfs = generate_cone(target, 1, 0, 181, Get_Angle(user, target.loc))
-			for(var/turf/checked_turf AS in cone_turfs)
-				for(var/mob/living/victim in checked_turf)
-					victim.flamer_fire_act(10)
-					victim.apply_damage(max(0, 20 - 20*victim.hard_armor.getRating("fire")), BURN, user.zone_selected, FIRE)
 
 		if(/datum/reagent/medicine/bicaridine)
-			if(isxeno(target))
-				return
-			INVOKE_ASYNC(src, PROC_REF(attack_async), source, target, user, weapon)
-			. = COMPONENT_ITEM_NO_ATTACK
+			INVOKE_ASYNC(src, PROC_REF(attack_bicaridine), source, target, user, weapon)
+			if(!(isxeno(target)))
```

Comment:
this sohuld be an intent lock or something, as it means you can't attack hostile erts etc.